lib/app/dispatcher.rb

```diff
@@ -86,7 +86,10 @@ def start_polling_jobs!
         scheduler.when_triggered do |connector_settings, task|
           case task
           when :sync
-            # update connector sync_now flag
+            # TODO: #update_connector_sync_now should be moved to Core::ConnectorSettings,
+            # there should not be any business logic related code in Core::ElasticConnectorActions.
+            # #update_connector_sync_now should not update `last_synced` after https://github.com/elastic/enterprise-search-team/issues/3366 is resolved,
+            # schedule should not based on `last_synced`
             Core::ElasticConnectorActions.update_connector_sync_now(connector_settings.id, false)
 
             Core::Jobs::Producer.enqueue_job(job_type: :sync, connector_settings: connector_settings)
```

Comment:
`update_connector_sync_now` won't reset `sync_now` flag in `connector_settings`

Reply:
Should we move the method to ConnectorSettings then?

Reply:
I do have the plan, as stated in the TODO comment. But not for 8.6